WordPress/src/test/java/org/wordpress/android/viewmodel/pages/PageListViewModelTest.kt

```diff
@@ -296,14 +300,95 @@ class PageListViewModelTest : BaseUnitTest() {
         assertThat((result[0].first[1] as Page).showOverlay).isEqualTo(false)
     }
 
+    @Test
+    fun `CANCEL_AUTO_UPLOAD is added to PublishedPage if auto upload is pending`() {
+        // Arrange
+        whenever(pageItemUiStateHelper.canCancelPendingAutoUpload(any(), any())).thenReturn(true)
+        whenever(pageItemUiStateHelper.setupPageActions(any(), any(), any())).thenCallRealMethod()
+
+        val pages = MutableLiveData<List<PageModel>>()
+        whenever(pagesViewModel.pages).thenReturn(pages)
+
+        viewModel.start(PUBLISHED, pagesViewModel)
+        val result = mutableListOf<Pair<List<PageItem>, Boolean>>()
+        viewModel.pages.observeForever { result.add(it) }
+
+        // Act
+        pages.value = listOf(buildPageModel(0))
+
+        // Assert
+        assertThat((result[0].first[0] as PublishedPage).actions).contains(CANCEL_AUTO_UPLOAD)
+    }
+
+    @Test
+    fun `CANCEL_AUTO_UPLOAD is not added to PublishedPage if auto upload is not pending`() {
+        // Arrange
+        whenever(pageItemUiStateHelper.canCancelPendingAutoUpload(any(), any())).thenReturn(false)
+        whenever(pageItemUiStateHelper.setupPageActions(any(), any(), any())).thenCallRealMethod()
+
+        val pages = MutableLiveData<List<PageModel>>()
+        whenever(pagesViewModel.pages).thenReturn(pages)
+
+        viewModel.start(PUBLISHED, pagesViewModel)
+        val result = mutableListOf<Pair<List<PageItem>, Boolean>>()
+        viewModel.pages.observeForever { result.add(it) }
+
+        // Act
+        pages.value = listOf(buildPageModel(0))
+
+        // Assert
+        assertThat((result[0].first[0] as PublishedPage).actions).doesNotContain(CANCEL_AUTO_UPLOAD)
+    }
+
+    @Test
+    fun `CANCEL_AUTO_UPLOAD is added to DraftPage if auto upload is pending`() {
+        // Arrange
+        whenever(pageItemUiStateHelper.canCancelPendingAutoUpload(any(), any())).thenReturn(true)
+        whenever(pageItemUiStateHelper.setupPageActions(any(), any(), any())).thenCallRealMethod()
+
+        val pages = MutableLiveData<List<PageModel>>()
+        whenever(pagesViewModel.pages).thenReturn(pages)
+
+        viewModel.start(DRAFTS, pagesViewModel)
+        val result = mutableListOf<Pair<List<PageItem>, Boolean>>()
+        viewModel.pages.observeForever { result.add(it) }
+
+        // Act
+        pages.value = listOf(buildPageModel(0, status = DRAFT))
+
+        // Assert
+        assertThat((result[0].first[0] as DraftPage).actions).contains(CANCEL_AUTO_UPLOAD)
+    }
+
+    @Test
+    fun `CANCEL_AUTO_UPLOAD is not added to DraftPage if auto upload is not pending`() {
+        // Arrange
+        whenever(pageItemUiStateHelper.canCancelPendingAutoUpload(any(), any())).thenReturn(false)
+        whenever(pageItemUiStateHelper.setupPageActions(any(), any(), any())).thenCallRealMethod()
```

Comment:
I am using the `thenCallRealMethod()` method so that I can verify that the logic within the helper is generating the appropriate actions depending on the `canCancelPendingAutoUpload`. I did a search of the codebase and realized it isn't used anywhere else so I am just bringing it your attention to hear your thoughts 😄

Reply:
If we call the real method it won't be a unit tests anymore. Since it'll test two separate units. 

>so that I can verify that the logic within the helper is generating the appropriate actions depending on the canCancelPendingAutoUpload

This would ideally be tested in unit tests for the helper. If it's tested there we can rely on it here and we can just mock it.

`thenCallRealMethod` is usually used for utility methods about which we are 99.99% sure they work as expected. Eg. when we use java Date methods etc. Chances that our code discovers a bug in Java is so low that it's ok to use `thenCallRealMethod` for such cases.

Reply:
Okay, thank you for this! I knew there was something I was missing and your explanation helps me to understand this clearly!

Reply:
Created separate unit tests for the actions use case adfffae and one for the page list 3cbea5b 

N.B in a future commit the `uploadUiState` is being passed to the component so the most updated tests are here e036fa0

Reply:
👍 Awesome! Btw thank you for asking these great questions! ;)